### ISSUE TITLE
Plans for ES2015+ native bundles?

### ISSUE THREAD

**Issue by mismith** [vuejs-templates/webpack#935]
As [this recently viral article](https://philipwalton.com/articles/deploying-es2015-code-in-production-today/) outlines, most modern browsers are now capable of handling ES2015+ syntax natively, meaning the Babel transpilation (and resulting code bloat) is not strictly necessary for many users.

How likely/possible/agreeable would be it be implement `<script type="module">` support along with `dist/app.min.js` + `dist/app.es5.min.js` bundle splitting?

**Comment by LinusBorg** [comment]
Those are two different things:

1. Transpilation
2. using native ES modules.

Concering *1.*, you can simply adjust your babel config to only support evergreen browsers. I don't think we will remove babel from this template as it's still required for many other things like JSX, object rest spread, decorators etc.

About *2.*, there are no plans yet. Since we recently started discussions about [vue-cli 3](https://github.com/vuejs/vue-cli/issues/589), I think this would be something we could solve through a preset for v3, but not in this template here.

**Comment by mismith** [comment]
Yeah, I guess these could be seen as two parts of the same end goal. As you mention, it's currently totally possible to target only the latest browsers, but like the article mentions, the key part is doing both at the same time, much like a progressive enhancement: new browsers use the latest and greatest, old browsers use the transpiled equivalent.

`vue-cli 3` sounds exciting, and maybe it is the best place for this kind of functionality to be added. I figured it would have to start at the template level instead of the cli tool level though, since this workflow would likely use webpack, and currently not all templates do.

What can I do next? Should this be discussed further or would it just need to gain popularity as an idea before anything happens?

**Comment by exarus** [comment]
@mismith the browsers can't resolve such expression:
`import _ from 'lodash'`
Because they don't know anything about `node_modules` and actually non-relative non-URL paths are currently not implemented.
